tests/components/alexa_devices/test_init.py

```diff
@@ -32,30 +37,58 @@ async def test_device_info(
     assert device_entry == snapshot
 
 
+@pytest.mark.parametrize(
+    ("minor_version", "country", "site_data", "site_login_data"),
+    [
+        # Standard migration case
+        (1, "US", None, None),
+        # Edge case #1: no country, site already in login data
+        (1, None, None, "https://www.amazon.com"),
+        # Edge case #2: no country, site in data (wrong place), minor version 1
+        (1, None, "https://www.amazon.com", None),
+        # Edge case #3: no country, site in data (wrong place), minor version 2
+        (2, None, "https://www.amazon.com", None),
+    ],
+)
 async def test_migrate_entry(
     hass: HomeAssistant,
     mock_amazon_devices_client: AsyncMock,
     mock_config_entry: MockConfigEntry,
+    minor_version: int,
+    country: str | None,
+    site_data: str | None,
+    site_login_data: str | None,
 ) -> None:
     """Test successful migration of entry data."""
+
+    data = {
+        CONF_USERNAME: TEST_USERNAME,
+        CONF_PASSWORD: TEST_PASSWORD,
+        CONF_LOGIN_DATA: {"session": "test-session"},
+    }
+
+    if country:
+        data[CONF_COUNTRY] = country
+
+    if site_data:
+        data[CONF_SITE] = site_data
+
+    if site_login_data:
+        data[CONF_LOGIN_DATA].update({CONF_SITE: site_login_data})
```

Comment:
Let's not have if statements in the code, I think it might be better to put the data in the paramtetrization